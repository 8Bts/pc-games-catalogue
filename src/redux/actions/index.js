import API from '../../api';

export const fetchSingleGameData = (id) => ({
  type: 'FETCH_SINGLE_GAME_DATA',
  payload: API.getSingleGameData(id),
});

export const fetchGamesData = () => ({
  type: 'FETCH_GAMES_DATA',
  payload: API.getAllGames(),
});

export const fetchGamesByGenre = (genre, page) => ({
  type: 'FETCH_GAMES_BY_GENRE',
  payload: API.getGamesByGenre(genre, page),
});

export const setGenre = (genre) => ({
  type: 'SET_GENRE',
  genre,
});

export const setPage = (page) => ({
  type: 'SET_CURRENT_PAGE',
  page,
});
