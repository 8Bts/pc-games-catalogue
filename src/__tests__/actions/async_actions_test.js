import configureMockStore from 'redux-mock-store';
import promise from 'redux-promise-middleware';
import nock from 'nock';
import * as Action from '../../redux/actions/index';

const middlewares = [promise];
const mockStore = configureMockStore(middlewares);

describe('async actions', () => {
  let store;
  beforeEach(() => {
    store = mockStore({});
  });
  afterEach(() => {
    nock.cleanAll();
  });

  it('creates FETCH_GAMES_DATA_FULFILLED when fetching games has been done', () => {
    nock('https://api.rawg.io')
      .defaultReplyHeaders({
        'access-control-allow-origin': '*',
        'access-control-allow-credentials': 'true',
      })
      .get('/api/games?key=0c2599def9e140fa9a93175f976efda3&page=1&page_size=15')
      .reply(200, []);

    return store.dispatch(Action.fetchGamesData())
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });

  it('creates FETCH_SINGLE_GAME_DATA_FULFILLED when fetching games has been done', () => {
    nock('https://api.rawg.io')
      .defaultReplyHeaders({
        'access-control-allow-origin': '*',
        'access-control-allow-credentials': 'true',
      })
      .get('/api/games/1?key=0c2599def9e140fa9a93175f976efda3')
      .reply(200, []);

    return store.dispatch(Action.fetchSingleGameData(1))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });

  it('creates FETCH_GAMES_BY_GENRE_FULFILLED when fetching games has been done', () => {
    nock('https://api.rawg.io')
      .defaultReplyHeaders({
        'access-control-allow-origin': '*',
        'access-control-allow-credentials': 'true',
      })
      .get('/api/games?key=0c2599def9e140fa9a93175f976efda3&page_size=15&genres=racing&page=1')
      .reply(200, []);

    return store.dispatch(Action.fetchGamesByGenre('racing'))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});
