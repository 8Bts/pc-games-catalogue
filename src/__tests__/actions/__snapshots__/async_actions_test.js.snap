// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async actions creates FETCH_GAMES_BY_GENRE_FULFILLED when fetching games has been done 1`] = `
Array [
  Object {
    "type": "FETCH_GAMES_BY_GENRE_PENDING",
  },
  Object {
    "payload": Array [],
    "type": "FETCH_GAMES_BY_GENRE_FULFILLED",
  },
]
`;

exports[`async actions creates FETCH_GAMES_DATA_FULFILLED when fetching games has been done 1`] = `
Array [
  Object {
    "type": "FETCH_GAMES_DATA_PENDING",
  },
  Object {
    "payload": Array [],
    "type": "FETCH_GAMES_DATA_FULFILLED",
  },
]
`;

exports[`async actions creates FETCH_SINGLE_GAME_DATA_FULFILLED when fetching games has been done 1`] = `
Array [
  Object {
    "type": "FETCH_SINGLE_GAME_DATA_PENDING",
  },
  Object {
    "payload": Array [],
    "type": "FETCH_SINGLE_GAME_DATA_FULFILLED",
  },
]
`;
